pkg/kubectl/cmd/util/factory.go

```diff
@@ -637,8 +637,8 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 				if t.Spec.Paused {
 					return true, nil
 				}
-				t.Spec.Paused = true
-				_, err := c.Extensions().Deployments(t.Namespace).Update(t)
+				body := []byte("{\"spec\":{\"paused\":true}}")
```

Comment:
Since this is version dependent, it needs to be in a positive list of API versions.  You need to have a switch block / if statement:

```
if API version not in [x, y, z] {
  return fmt.Errorf("API version is not recognized, cannot apply patch')
}
```

and you need to have good testing around it to ensure this works with all API versions.